src/server/rpc/procedures/google-maps/google-maps.js

```diff
@@ -4,37 +4,30 @@
 // size of the image for each user in the given group
 'use strict';
 
-var debug = require('debug'),
-    trace = debug('netsblox:rpc:static-map:trace'),
-    request = require('request'),
+const ApiConsumer = require('../utils/api-consumer'),
     SphericalMercator = require('sphericalmercator'),
-    merc = new SphericalMercator({size:256}),
-    CacheManager = require('cache-manager'),
     Storage = require('../../storage'),
-    // TODO: Change this cache to mongo or something (file?)
-    // This cache is shared among all StaticMap instances
-    cache = CacheManager.caching({store: 'memory', max: 1000, ttl: Infinity}),
-    key = process.env.GOOGLE_MAPS_KEY;
+    merc = new SphericalMercator({size:256}),
+    KEY = process.env.GOOGLE_MAPS_KEY;
+
+let GoogleMap = new ApiConsumer('staticmap', 'https://maps.googleapis.com/maps/api/staticmap?');
 
-// TODO: check that the env variable is defined
 var storage;
 
-// Retrieving a static map image
-var baseUrl = 'https://maps.googleapis.com/maps/api/staticmap',
-    getStorage = function() {
-        if (!storage) {
-            storage = Storage.create('static-map');
-        }
-        return storage;
-    };
+GoogleMap._state = {};
+GoogleMap._state.userMaps = {};
+GoogleMap._state.roomId = {};
```

Comment:
`roomId` isn't set anywhere... This will be a problem because every room will be colliding...